src/redux/ensRegistration.ts

```diff
@@ -0,0 +1,301 @@
+import { omit } from 'lodash';
+import { AppDispatch, AppGetState } from './store';
+import {
+  ENSRegistrationState,
+  EthereumAddress,
+  Records,
+  RegistrationParameters,
+} from '@rainbow-me/entities';
+
+const ENS_REGISTRATION_UPDATE_DURATION =
+  'ensRegistration/ENS_REGISTRATION_UPDATE_DURATION';
+const ENS_REGISTRATION_UPDATE_RECORDS =
+  'ensRegistration/ENS_REGISTRATION_UPDATE_RECORDS';
+const ENS_REGISTRATION_UPDATE_RECORD_BY_KEY =
+  'ensRegistration/ENS_REGISTRATION_UPDATE_RECORD_BY_KEY';
+const ENS_REGISTRATION_REMOVE_RECORD_BY_KEY =
+  'ensRegistration/ENS_REGISTRATION_REMOVE_RECORD_BY_KEY';
+const ENS_CONTINUE_REGISTRATION = 'ensRegistration/ENS_CONTINUE_REGISTRATION';
+const ENS_START_REGISTRATION = 'ensRegistration/ENS_START_REGISTRATION';
+const ENS_SAVE_COMMIT_REGISTRATION_PARAMETERS =
+  'ensRegistration/ENS_SAVE_COMMIT_REGISTRATION_PARAMETERS';
+
+interface EnsRegistrationUpdateDurationAction {
+  type: typeof ENS_REGISTRATION_UPDATE_DURATION;
+  payload: ENSRegistrationState;
+}
+
+interface EnsRegistrationUpdateRecordsAction {
+  type: typeof ENS_REGISTRATION_UPDATE_RECORDS;
+  payload: ENSRegistrationState;
+}
+
+interface EnsRegistrationUpdateRecordByKeyAction {
+  type: typeof ENS_REGISTRATION_UPDATE_RECORD_BY_KEY;
+  payload: ENSRegistrationState;
+}
+
+interface EnsRegistrationRemoveRecordByKeyAction {
+  type: typeof ENS_REGISTRATION_REMOVE_RECORD_BY_KEY;
+  payload: ENSRegistrationState;
+}
+
+interface EnsRegistrationContinueRegistrationAction {
+  type: typeof ENS_CONTINUE_REGISTRATION;
+  payload: ENSRegistrationState;
+}
+
+interface EnsRegistrationStartRegistrationAction {
+  type: typeof ENS_START_REGISTRATION;
+  payload: ENSRegistrationState;
+}
+
+interface EnsRegistrationSaveCommitRegistrationParametersAction {
+  type: typeof ENS_SAVE_COMMIT_REGISTRATION_PARAMETERS;
+  payload: ENSRegistrationState;
+}
+
+export type EnsRegistrationActionTypes =
+  | EnsRegistrationUpdateDurationAction
+  | EnsRegistrationUpdateRecordsAction
+  | EnsRegistrationUpdateRecordByKeyAction
+  | EnsRegistrationRemoveRecordByKeyAction
+  | EnsRegistrationContinueRegistrationAction
+  | EnsRegistrationStartRegistrationAction
+  | EnsRegistrationSaveCommitRegistrationParametersAction;
+
+// -- Actions ---------------------------------------- //
+export const startRegistration = (
+  accountAddress: EthereumAddress,
+  name: string
+) => async (dispatch: AppDispatch, getState: AppGetState) => {
+  const {
+    ensRegistration: { registrations },
+  } = getState();
+
+  const lcAccountAddress = accountAddress.toLowerCase();
```

Comment:
Just FYI: In general, we tend to use toLowerCase() because it checks for null values
Anyway, In this case if accountAddress is null we'd have much bigger problems 😅

Reply:
hm is it null sometimes? I took this pattern from the nonce manager 🤔